lib/src/chart/bar_chart/bar_chart_painter.dart

```diff
@@ -226,7 +226,20 @@ class BarChartPainter extends AxisChartPainter<BarChartData> with TouchHandler<B
                 bottomRight: borderRadius.bottomRight);
           }
 
-          _barPaint.color = barRod.backDrawRodData.color;
+          if (barRod.colors.length > 1) {
+            final rect = Rect.fromLTRB(

```

Comment:
You can define an extension function on RRect class in **src/extensions** directory,
It makes the code more readable, and you can reuse it in the next section too.